.github/workflows/doc-tests.yml

```diff
@@ -11,6 +11,10 @@ on:
   pull_request:
     branches: [ master, main-* ]
 
+concurrency:
+  group: unit-tests-${{ github.event.pull_request.number || github.ref }}
```

Comment:
The group name prefix `unit-tests` confuses me. It looks like you're defining a single group for anything that refers to the same commit, so `commit-` instead of `unit-tests-` might be more straightforward (Do we even need `github.event.pull_request.number || github.ref` instead of `github.ref` ?)

Reply:
Fixed up these matters in a followup PR